job_creator.php

```diff
@@ -0,0 +1,273 @@
+<?php
+
+class JobCreator
+{
+    private $installerVersion = null;
+
+    /**
+     * Get the correct version of silverstripe/installer to include for the given repository and branch
+     */
+    public function getInstallerVersion(string $githubRepository, string $branch): string
+    {
+        $repo = explode('/', $githubRepository)[1];
+        if (in_array($repo, NO_INSTALLER_REPOS)) {
+            return '';
+        }
+        // e.g. ['4', '11']
+        $portions = explode('.', $branch);
+        if (count($portions) == 1) {
+            return '4.x-dev';
+        }
+        if (in_array($repo, LOCKSTEPED_REPOS)) {
+            return '4' . $portions[1] . 'x-dev';
+        } else {
+            // use the latest minor version of installer
+            $installerVersions = array_keys(INSTALLER_TO_PHP_VERSIONS);
+            // remove '4' version
+            $installerVersions = array_diff($installerVersions, ['4']);
+            // get the minor portions of the verisons e.g. [9, 10, 11]
+            $minorPortions = array_map(fn($portions) => (int) explode('.', $portions)[1], $installerVersions);
+            sort($minorPortions);
+            return $minorPortions[count($minorPortions) - 1] . '.x-dev';
```

Comment:
Won't this return `10.x-dev`, `11.x-dev`, etc instead of `4.10.x-dev`, etc?

Reply:
Opps :-) Removed the '4.' prefix when I refactored